test/e2e/basic_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"log"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+)
+
+func convertStdoutToString(stdout io.ReadCloser) string {
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(stdout)
+	return buf.String()
+}
+
+func runCmdAndReturnStdout(t *testing.T, cmd string, cmdArgs ...string) string {
+	statusCmd := exec.Command(cmd, cmdArgs...)
+	stdout, err := statusCmd.StdoutPipe()
+	assert.Equal(t, err, nil)
+	err = statusCmd.Start()
+	assert.Equal(t, err, nil)
+	return convertStdoutToString(stdout)
+}
+
+func getJobStatus(t *testing.T, statusStr string, result map[string]interface{}) string {
+	json.Unmarshal([]byte(statusStr), &result)
+	statusField := result["status"].(map[string]interface{})["applicationState"]
+	errMsg := statusField.(map[string]interface{})["errorMessage"].(string)
+	assert.Equal(t, errMsg, "")
+	return statusField.(map[string]interface{})["state"].(string)
+}
+
+func TestSubmitSparkPiYaml(t *testing.T) {
+	t.Parallel()
+
+	// Wait for test job to finish. Time out after 90 seconds.
+	timeout := 90
+
+	yamlPath := "../../examples/spark-pi.yaml"
+	submitResult := runCmdAndReturnStdout(t, "kubectl", "apply", "-f", yamlPath)
```

Comment:
This assumes that the machine running the test must have `kubectl` installed and present on `PATH`. Can we instead use the generated client to create the resource?